tests/rescoup/test_chopstep.cpp

```diff
@@ -0,0 +1,291 @@
+            );
+            this->exit_code_ = this->flow_main_->executeInitStep();
+            if (this->exit_code_ == EXIT_SUCCESS) {
+                this->simulator_ = this->flow_main_->getSimulatorPtr();
+            }
+        }
+    }
+    Simulator* getSimulator() { return simulator_; }
+
+private:
+    int exit_code_;
+    FlowMainPtr flow_main_;
+    Simulator *simulator_;
+};
+
+} // namespace Opm
+
+namespace {
+class OpmSimulatorTestCase
+{
+public:
+    using MainTestWrapper = Opm::MainTestWrapper;
+    using Simulator = Opm::MainTestWrapper::Simulator;
+
+    OpmSimulatorTestCase(
+        const std::string& prog_name,
+        const std::string& deck_filename,
+        const std::vector<std::string>& args
+    )
+    {
+        this->save_argv_.push_back(prog_name);
+        this->save_argv_.insert(save_argv_.end(), args.begin(), args.end());
+        this->save_argv_.push_back(deck_filename);
+        this->argv_.reserve(this->save_argv_.size() + 1);  // +1 for nullptr
+        for (const auto& arg : this->save_argv_) {
+            this->argv_.push_back(const_cast<char*>(arg.c_str()));
+        }
+        // Make sure the last element is nullptr, this is required by MPI_Init()
+        this->argv_.push_back(nullptr);
+        char **argv = this->argv_.data();
+        int argc = this->argv_.size() - 1;
+        this->main_ = std::make_unique<MainTestWrapper>(argc, argv);
+    }
+
+    MainTestWrapper& getMain()
+    {
+        return *main_;
+    }
+
+    Simulator* getSimulatorPtr()
+    {
+        return main_->getSimulator();
+    }
+
+    private:
+        std::unique_ptr<MainTestWrapper> main_;
+        std::vector<std::string> save_argv_;
+        std::vector<char *> argv_;
+};
+
+struct SimulatorFixture
+{
+    using Simulator = Opm::MainTestWrapper::Simulator;
+    SimulatorFixture() :
+        simulator_wrapper{
+            /*prog_name=*/"flow_test_chopstep",
+            /*deck_filename=*/"RC-01_MAST_PRED.DATA",
+            /*args=*/{"--parsing-strictness=low"}  // Need this until PR #5643 is merged
+        },
+        simulator{simulator_wrapper.getSimulatorPtr()},
+        schedule{simulator->vanguard().schedule()},
+        rc_master{Opm::FlowGenericVanguard::comm(), schedule, 0, nullptr},
+        start_date{static_cast<double>(schedule.getStartTime())}
+    {
+        rc_master.addSlaveName("RES-1");
+        rc_master.addSlaveName("RES-2");
+        rc_master.resizeSlaveStartDates(2);
+        rc_master.resizeNextReportDates(2);
+    }
+
+    // Sets the slave start date and the next report time offset for the given slave number
+    // NOTE: The method name is kept short as setSlaveDate() instead of a longer descriptive name
+    //   like setSlaveStartDateAndNextReportTimeOffset() because the method is called multiple times
+    //   in the test cases and the shorter name makes the test cases more readable.
+    void setSlaveData(int slave_number, double slave_start_date, double report_time_step_size)
```

Comment:
I kind of disagree. The reason that I asked for a name change is such that fellow developers do not need to take a look at the documentation to understand what is going on. IMHO a short name that does not reveal anything does not make code more readable.

Reply:
make a struct, document the struct members, pass an instance of that struct, everybody wins.

Reply:
>The reason that I asked for a name change is such that fellow developers do not need to take a look at the documentation to understand what is going on.

@blattms In general I would agree, but in this case the function is called multiple times and the long name could become distracting (reducing readability).